plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -333,6 +483,46 @@ private Boolean isModelNotDeployed(MLModelState mlModelState) {
             && !mlModelState.equals(MLModelState.PARTIALLY_DEPLOYED);
     }
 
+    private <T> Boolean isPipelineContainsModel(
+        List<T> pipelineConfigurations,
+        String candidateModelId,
+        Function<T, Map<String, Object>> getConfigFunction
+    ) {
+        for (T pipelineConfiguration : pipelineConfigurations) {
+            Map<String, Object> config = getConfigFunction.apply(pipelineConfiguration);
+            if (searchThroughConfig(config, candidateModelId, "")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Boolean searchThroughConfig(Object searchCandidate, String candidateId, String prefixKey) {
```

Comment:
Is that correct the prefixKey is the leaf target key looking for? And here it's a constant value `model_id`? If so, two suggestions are:
1. Make "model_id" a private static value with a meaningful name like `TARGET_MODEL_KEY`.
2. Change the prefixKey to a self explanation name like `targetModelKey`.

Reply:
Done